File: NetFramework/XHtmlEditor/VB/CustomDialogs/Controls/FindOptions.vb
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing
Imports System.Data
Imports System.Text
Imports System.Windows.Forms

Friend Partial Class FindOptions
	Inherits UserControl
    Public Sub New()
        Me.components = Nothing
        Me.InitializeComponent()
        AddHandler Me._btnClose.Click, AddressOf CloseClick
        AddHandler Me._btnFindNext.Click, AddressOf FindNextClick

        Me._tbFindPattern_TextChanged(Me, EventArgs.Empty)
    End Sub


    Sub FindNextClick(ByVal sender As Object, ByVal e As EventArgs)
        Me.RaiseFindNextButonClick()
    End Sub

    Sub CloseClick(ByVal sender As Object, ByVal e As EventArgs)
        Me.RaiseCloseButtonClick()
    End Sub




    Private Sub _tbFindPattern_TextChanged(ByVal sender As Object, ByVal e As EventArgs)
        _btnFindNext.Enabled = SearchPattern.Length > 0
    End Sub

#Region "events"

    Public Event ButtonFindNextClick As EventHandler

    Protected Sub RaiseFindNextButonClick()
        RaiseEvent ButtonFindNextClick(Me, EventArgs.Empty)
    End Sub

    Public Event ButtonCloseClick As EventHandler

    Protected Sub RaiseCloseButtonClick()
        RaiseEvent ButtonCloseClick(Me, EventArgs.Empty)
    End Sub

#End Region

#Region "properties"

    Public Property MatchCase() As Boolean
        Get
            Return _chbMatchCase.Checked
        End Get
        Set(ByVal value As Boolean)
            _chbMatchCase.Checked = value
        End Set
    End Property

    Public Property MatchWholeWord() As Boolean
        Get
            Return _chbMatchWholeWord.Checked
        End Get
        Set(ByVal value As Boolean)
            _chbMatchWholeWord.Checked = value
        End Set
    End Property

    Public Property SearchUp() As Boolean
        Get
            Return _chbSearchUp.Checked
        End Get
        Set(ByVal value As Boolean)
            _chbSearchUp.Checked = value
        End Set
    End Property

    Public Property SearchPattern() As String
        Get
            Return _tbFindPattern.Text.Trim()
        End Get
        Set(ByVal value As String)
            _tbFindPattern.Text = value
        End Set
    End Property

    Public ReadOnly Property AcceptButton() As Button
        Get
            Return _btnFindNext
        End Get
    End Property

    Public ReadOnly Property CancelButton() As Button
        Get
            Return _btnClose
        End Get
    End Property

#End Region

#Region "public"

    Public Sub SetActiveControl()
        _tbFindPattern.[Select]()
    End Sub

#End Region
End Class

File: NetFramework/XHtmlEditor/CS/CustomDialogs/Controls/FindOptions.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Text;
using System.Windows.Forms;

namespace CustomDialogs
{
    internal partial class FindOptions : UserControl
    {
        public FindOptions()
        {
            InitializeComponent();

            _btnClose.Click += delegate(object sender, EventArgs e) { RaiseCloseButtonClick(); };
            _btnFindNext.Click += delegate(object sender, EventArgs e) { RaiseFindNextButonClick(); };

            _tbFindPattern_TextChanged(this, EventArgs.Empty);
        }

        private void _tbFindPattern_TextChanged(object sender, EventArgs e)
        {
            _btnFindNext.Enabled = SearchPattern.Length > 0;
        }

        #region events

        public event EventHandler ButtonFindNextClick;

        protected void RaiseFindNextButonClick()
        {
            if (ButtonFindNextClick != null)
                ButtonFindNextClick(this, EventArgs.Empty);
        }

        public event EventHandler ButtonCloseClick;

        protected void RaiseCloseButtonClick()
        {
            if (ButtonCloseClick != null)
                ButtonCloseClick(this, EventArgs.Empty);
        }

        #endregion

        #region properties

        public bool MatchCase
        {
            get { return _chbMatchCase.Checked; }
            set { _chbMatchCase.Checked = value; }
        }

        public bool MatchWholeWord
        {
            get { return _chbMatchWholeWord.Checked; }
            set { _chbMatchWholeWord.Checked = value; }
        }

        public bool SearchUp
        {
            get { return _chbSearchUp.Checked; }
            set { _chbSearchUp.Checked = value; }
        }

        public string SearchPattern
        {
            get { return _tbFindPattern.Text.Trim(); }
            set { _tbFindPattern.Text = value; }
        }

        public Button AcceptButton
        {
            get { return _btnFindNext; }
        }

        public Button CancelButton
        {
            get { return _btnClose; }
        }

        #endregion

        #region public

        public void SetActiveControl()
        {
            _tbFindPattern.Select();
        }

        #endregion
    }
}
